src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -34,7 +34,14 @@ TransactionManager::TransactionManager(
     lba_manager(std::move(_lba_manager)),
     journal(std::move(_journal)),
     epm(std::move(_epm)),
-    backref_manager(std::move(_backref_manager))
+    backref_manager(std::move(_backref_manager)),
+#ifdef UNIT_TESTS_BUILT
```

Comment:
> I think we should be constrained to use the macro `UNIT_TESTS_BUILT` -- only use it when inevitable.
> 
> And is it better to create a test profile to manage the configurations in case of UTs in a central place, rather than specializing it everywhere? Same to `seastar_max_data_allocation_size` in this PR.

I added a test profile for the full integrity check. Since some test cases needs `seastar_max_data_allocation_size` disabled, while others are not, I can't implement a test profile for all the cases, so the `disable/enable_max_extent_size()` interface is preserved.